### ISSUE TITLE
JES Preemption during scatter operation leads to stall after scatter block

### ISSUE THREAD

**Issue by patmagee** [broadinstitute/cromwell#1690]
When running some of our pipelines with the latest release version (0.22), I am not able to get the pipelines to actually complete if they run into a preemption during their scatter operation. The scatter block will successfully complete for all cases, however the execution stalls and does not progress any further.

To make sure this had nothing to do with a slow process I let the job sit overnight, and still it did not complete, or even move past the "Success" of the final scatter operation to complete. I used the wdl code below to replicate the issue several times locally, without having to run our entire workflow

```
task c {
   Int num
   command {
	echo ${num}
        #Wait 5 minutes, and hope for a preemption. make longer to force preemption
	sleep 300
   }
   runtime {
     docker: "ubuntu"
     preemptible: 4
   }
   output {
     Int out = read_int(stdout())
   }
}
task d {
   Array[Int] num
   command {
	echo ${sep=" " num}
        sleep 300
   }
   runtime {
     docker: "ubuntu"
     preemptible: 4
   }
   output {
     String out=stdout()
   }
}
workflow wf {
 Array[Int] first = [1,2,3,4,5]

 scatter (x in first){
    	call c {input: num=x}
	call c as a {input: num=c.out}
 }

 call d {input: num=a.out}
 output {
    d.*
 }
}
```

**Comment by ruchim** [comment]
[wf_metadata.txt](https://github.com/broadinstitute/cromwell/files/601036/wf_metadata.txt)

@patmagee  I was able to reproduce this issue and will start investigating.

**Comment by patmagee** [comment]
@ruchim Just fyi, the workflows ran fine with preemptions occuring in 0.21 release

**Comment by ruchim** [comment]
Fixed by #1697
